src/Microsoft.Identity.Client/Http/HttpClientFactory.cs

```diff
@@ -27,32 +27,31 @@
 
 using System.Net.Http;
 using System.Net.Http.Headers;
+using Microsoft.Identity.Client.AppConfig;
 
 namespace Microsoft.Identity.Client.Http
 {
-    internal interface IHttpClientFactory
+    internal class HttpClientFactory : IMsalHttpClientFactory
     {
-        HttpClient HttpClient { get; }
-    }
+        private readonly HttpClient _httpClient;
 
-    internal class HttpClientFactory : IHttpClientFactory
-    {
         // The HttpClient is a singleton per ClientApplication so that we don't have a process wide singleton.
         public const long MaxResponseContentBufferSizeInBytes = 1024*1024;
 
         public HttpClientFactory()
         {
-            var httpClient = new HttpClient(new HttpClientHandler() { UseDefaultCredentials = true })
+            _httpClient = new HttpClient(new HttpClientHandler() { UseDefaultCredentials = true })
```

Comment:
>_httpClient [](start = 12, length = 11)

The httpClient should be possible specified externally (e.g. on the config object). Not sure that is the PR, though?

Reply:
We already have that:

public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)

This class is used if the developer doesn't specify their own HttpClientFactory

---
In reply to: [247265123](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/780#discussion_r247265123) [](ancestors = 247265123)